fem/quadinterpolator.cpp

```diff
@@ -420,6 +420,34 @@ void QuadratureInterpolator::MultTranspose(
    MFEM_ABORT("this method is not implemented yet");
 }
 
+static void D2QValues1D(const int NE,
+                        const Array<double> &b_,
+                        const Vector &x_,
+                        Vector &y_,
+                        const int vdim = 1,
+                        const int d1d = 0,
+                        const int q1d = 0)
+{
+   auto b = Reshape(b_.Read(), q1d, d1d);
+   auto x = Reshape(x_.Read(), d1d, vdim, NE);
+   auto y = Reshape(y_.Write(), vdim, q1d, NE);
+
+   MFEM_FORALL(e, NE,
+   {
+      for (int c = 0; c < vdim; c++)
+      {
+         for (int q = 0; q < q1d; ++q)
+         {
```

Comment:
If performance matters a bit, I imagine that the following would be faster on GPU (maybe slower on CPU):
```suggestion
   MFEM_FORALL(i, NE*vdim*q1d,
   {
      const int vdim = i%(NE*q1d);
      const int e = i%(vdim*q1d);
      const int q = i%(vdim*NE);
```

Reply:
Too fancy for a 1D kernel; not done.